src/mame/drivers/twinkle.cpp

```diff
@@ -1047,8 +1054,7 @@ void twinkle_state::twinkle(machine_config &config)
 
 	M68000(config, m_audiocpu, 32000000/2);    /* 16.000 MHz */
 	m_audiocpu->set_addrmap(AS_PROGRAM, &twinkle_state::sound_map);
-	m_audiocpu->set_periodic_int(FUNC(twinkle_state::irq1_line_assert), attotime::from_hz(60));
-	m_audiocpu->set_periodic_int(FUNC(twinkle_state::irq2_line_assert), attotime::from_hz(60));
+	m_audiocpu->set_periodic_int(FUNC(twinkle_state::periodic_int), attotime::from_hz(60));
```

Comment:
This is a change I was planning on making eventually for Twinkle after my recent work with Firebeat so I think the change itself is fine, but you should be careful with the timing of the interrupts here. I haven't looked too hard at Twinkle yet but the SPU in Firebeat and Twinkle are extremely similar. You can find my notes on how the interrupts for the SPU program work for Firebeat here: https://github.com/mamedev/mame/blob/master/src/mame/drivers/firebeat.cpp#L1005-L1051

The key point about this is the interrupt that interacts with the RF5C400 (IRQ2 for Firebeat) is actually responsible for animating the volume, pan, etc so if the interrupt timing is too slow or too fast then it will have some undesired effects such as fade in/outs taking too long, or sounds not stopping at the appropriate times. The issues won't always manifest all of the time either. For example, in pop'n music Animelo 2, some keysounds would not be stopped in time resulting in a buzzing sound in some songs on only certain keysound samples when the interrupt was too slow.

I settled on 500hz for Firebeat after a lot of testing but it's not something I can back up with hard evidence. But I have a feeling 60hz is probably too slow for Twinkle too.

Reply:
I think 60hz might be slightly slow but it's around the right ballpark for twinkle, it feels kinda hard to compare to videos of real HW since the loading between menu's seems to be faster on mame. Since I didn't really know any better I just kept it at what it was originally (which I guess was probably pretty arbitrary in the first place though).

Reply:
Yeah, I think 60hz were maybe just a rough guess since all interrupt frequencies are set to that. I gave the interrupt code in the SPU program a quick look and it seems like does not execute 8 callbacks worth of commands like Firebeat, and instead does all of the sound animations in one go each time IRQ 1 is called (thank god). So 500hz would definitely be too high here.

I'm not comfortable giving code/style reviews for MAME PRs since I'm still learning myself so I'll leave that up to someone else, but I gave the PR a test and it fixes all of the issues I noticed recently (sounds not stopping, buzzing during the song loading screen where it shows you your options, etc) so the logic itself looks good to me.

I will be doing an SPU refactor in the near future so I will try tweaking the IRQ frequencies at that time if it's needed.